test/runtests.jl

```diff
@@ -68,3 +68,30 @@ end
         end
     end
 end
+
+@testset "empty tseries" begin
+    w = Workspace(; 
+        t1 = TSeries(1995Q1),
+        t2 = TSeries(1993Q3, Vector{Float32}()),
+        t3 = TSeries(1996Q2, Vector{Bool}()),
+        t4 = TSeries(1996Q2, Vector{Bool}([true, false, true]))
+    )
+    writefame("data.db", w)
```

Comment:
You can also use the work database for the tests, instead of writing a file on disk. `db = workdb()` gets you a handle, which you can pass instead of filename.  Just a suggestion. The only advantage is that using the work db you don't have to delete data.db after.